test/Parse/explicit_lifetime_dependence_specifiers.swift

```diff
@@ -0,0 +1,118 @@
+    self.ptr = ptr
+  }
+}
+
+func testBasic() {
+  let capacity = 4
+  let a = Array(0..<capacity)
+  a.withUnsafeBytes {
+    let view = BufferView($0)
+    let derivedView = derive(view)
+    let newView = consumeAndCreate(derivedView)
+    use(newView)    
+  }
+}
+
+func derive(_ x: borrowing BufferView) -> _borrow(x) BufferView {
+  return BufferView(x.ptr)
+}
+
+func consumeAndCreate(_ x: consuming BufferView) -> _copy(x) BufferView {
+  return BufferView(x.ptr)
+}
+
+func use(_ x: borrowing BufferView) {}
+
+func deriveMultiView1(_ x: borrowing BufferView, _ y: borrowing BufferView) -> _borrow(x, y) BufferView {
+  if (Int.random(in: 1..<100) % 2 == 0) {
+    return BufferView(x.ptr)
+  }
+  return BufferView(y.ptr)
+}
+
+func deriveMultiView2(_ x: borrowing BufferView, _ y: borrowing BufferView) -> _borrow(x) _borrow(y) BufferView {
+  if (Int.random(in: 1..<100) % 2 == 0) {
+    return BufferView(x.ptr)
+  }
+  return BufferView(y.ptr)
+}
+
+func consumeAndCreateMultiView1(_ x: consuming BufferView, _ y: consuming BufferView) -> _copy(x, y) BufferView {
+  if (Int.random(in: 1..<100) % 2 == 0) {
+    return BufferView(x.ptr)
+  }
+  return BufferView(y.ptr)
+}
+
+func consumeAndCreateMultiView2(_ x: consuming BufferView, _ y: consuming BufferView) -> _copy(x) _copy(y) BufferView {
+  if (Int.random(in: 1..<100) % 2 == 0) {
+    return BufferView(x.ptr)
+  }
+  return BufferView(y.ptr)
+}
+
+func mixedMultiView(_ x: consuming BufferView, _ y: borrowing BufferView) -> _copy(x) _borrow(y) BufferView {
+  if (Int.random(in: 1..<100) % 2 == 0) {
+    return BufferView(x.ptr)
+  }
+  return BufferView(y.ptr)
+}
+
+func modifiedViewDependsOnInput(_ x: inout MutableBufferView) -> _mutate(x) MutableBufferView {
+  return MutableBufferView(x.ptr)
+}
+
+func modifiedViewDependsOnParent(_ x: inout MutableBufferView) -> _copy(x) MutableBufferView {
+  return MutableBufferView(x.ptr)
+}
+
+func invalidSpecifier1(_ x: borrowing BufferView) -> _borrow BufferView { // expected-error{{expected '(' after lifetime dependence specifier}}
+  return BufferView(x.ptr)
+}
+
+func invalidSpecifier2(_ x: borrowing BufferView) -> _borrow() BufferView {// expected-error{{expected identifier or index or self in lifetime dependence specifier}}
+  return BufferView(x.ptr)
+}
+
+func invalidSpecifier3(_ x: borrowing BufferView) -> _borrow(*) BufferView { // expected-error{{expected identifier or index or self in lifetime dependence specifier}}
+  return BufferView(x.ptr)
+} 
+
+func invalidSpecifier4(_ x: borrowing BufferView) -> _borrow(0) BufferView {
+  return BufferView(x.ptr)
+}
+
+struct Wrapper : ~Escapable {
+  let view: BufferView
+  borrowing func getView1() -> _borrow(self) BufferView {
+    return view
+  }
+
+  consuming func getView2() -> _consume(self) BufferView {
+    return view
+  }
+
+  mutating func getView3() -> _copy(self) BufferView {
+    return view
+  }
+
+  borrowing func getView4() -> _copy(self) BufferView {
+    return view
+  }
+}
+
```

Comment:
Do we have a test case what happens if putting `_borrow(self)` on a variable or parameter? Ie.

```swift
var x: _borrow(self) BufferView
```

and 

```swift
func foo(_ x: _borrow(self) BufferView) {}
```

I’m not sure what the expected behavior should be but I think it would be good to cover this.

Reply:
This is in `Sema/explicit_lifetime_specifiers.swift`

```
func invalidSpecifierPosition(_ x: borrowing _borrow(x) BufferView) -> BufferView { // expected-error{{lifetime dependence specifiers may only be used on result of functions, methods, initializers}}
  return BufferView(x.ptr)
}
```

I'll add `var x: _borrow(self) BufferView` as well.